server/src/main/java/io/crate/planner/operators/HashAggregate.java

```diff
@@ -76,6 +77,12 @@ public ExecutionPlan build(DependencyCarrier executor,
                                @Nullable Integer pageSizeHint,
                                Row params,
                                SubQueryResults subQueryResults) {
+        // Ignore source look-ups optimization for aggregations because the field for source-lookups
```

Comment:
The primary reason here should be performance.  Global aggregations are a pipeline breaker using all data.  So column store is likely more efficient. 

That the implementation (currently) also can't use source lookups is secondary as its an implementation detail that shouldn't matter as much for the planner

(Hence also the assertion for the implementation.  It's not ideal that one far removed component relies on another component acting in a certain way,  but I think the trade-off is acceptable here. Otherwise we'd need to add the rewrite to non-source lookup you had in the initial version back too - only for it to be dead coded)

Reply:
ok 👍